core/src/worker/workflow/machines/workflow_machines.rs

```diff
@@ -420,9 +418,15 @@ impl WorkflowMachines {
                 Some(workflow_activation_job::Variant::QueryWorkflow(_))
             )
         });
+        let is_replaying = self.replaying || all_query;
+        let build_id_for_current_task = if is_replaying {
+            self.current_wft_build_id.clone().unwrap_or_default()
```

Comment:
Does core need the same logic as Go to check for an empty build ID?

Reply:
The lookahead works a bit differently and avoids this problem - I actually ended up changing the Go one to be a bit simpler and more similar in my last fix too.